src/traces/parcoords/parcoords.js

```diff
@@ -557,7 +557,7 @@ module.exports = function(root, svg, styledData, layout, callbacks) {
                     .outerTickSize(2)
                     .ticks(wantedTickCount, d.tickFormat) // works for continuous scales only...
                     .tickValues(d.ordinal ? // and this works for ordinal scales
-                        sdom.map(function(d, i) {return texts && texts[i] || d;}) :
+                        sdom.map(function(d, i) {return texts && texts[i] !== undefined ? texts[i] : d;}) :
```

Comment:
Ah, it still gets [here](https://github.com/plotly/plotly.js/blob/f42802689a99cba4057434664853e848c75625d6/src/plots/cartesian/axes.js#L1191) - which stringifies anything. This should definitely change... I think unfortunately both of these blocks are required in the cartesian case (~line 878 is for showing all ticks, ~line 1191 gets invoked more directly on hover) but yeah, would be great if they could share logic for excluding `undefined` and `null`.